src/parse.js

```diff
@@ -602,30 +647,26 @@ function parse(source, root, options) {
     }
 
     function parseOption(parent, token) {
-        var isCustom = skip("(", true);
-
-        /* istanbul ignore if */
-        if (!typeRefRe.test(token = next()))
-            throw illegal(token, "name");
-
-        var name = token;
-        var option = name;
+        var identifier = readOptionIdentifier();
+
+        // Historically, `(some.option).prop` has been interpreted as a property
```

Comment:
This sounds like exactly what it is today.  What's the change in behavior here?

Reply:
This is mostly a comment I found useful to explain why the somewhat strange code exists in the first place. Say, someone stumbles over this part, wonders why it's there, and decides to make a PR to remove it because it doesn't seem useful to them. Other than that, there's indeed no change in behavior other than also applying the fallback to syntax previously not supported, e.g. `a.(b.c).d`.

Reply:
Ah I think I misunderstood what was happening here.  FWIW it looks like you still don't support *multiple* extensions (e.g. `a.(b).c.(d).e`), but we hopefully won't ever need that